### ISSUE TITLE
Support Aeson 2.x and LTS 19?

### ISSUE THREAD

**Issue by unliftedq** [higherkindness/mu-haskell#340]
From this PR #332, I see 
`LTS 19 unfortunately doesn't work, due to the Aeson 2.x versioning.`
Could we upgrade Aeson to 2.x? Is there any hard dependency on Aeson 1.x?

**Comment by sfwc** [comment]
Has there been any progress on this recently? I'd like to use mu-haskell in a new project, but not having LTS 19 is turning out to be a dealbreaker.

Taking a stab at it myself, the issues with Aeson 2 seem relatively easy to fix (just `KeyMap` for objects, plus passing the `-json` build flag to the `graphql` package), but next I see an issue in `http2-client` (maybe GHC version-related?) that I'm not immediately sure what to do about.

**Comment by serras** [comment]
Not really. The main problem is indeed `http2-client`, which no longer compiles. Not only that, the package `http2` has added its own `Client` module which conflicts with that one :/

**Comment by listx** [comment]
> The problem is that some of the dependencies we use are not compatible with Aeson 1.x. But I can try again, maybe it works this time!

I think you meant to say 2.x?

Anyway, is there some way others can help out here? Are we blocked by https://github.com/haskell-grpc-native/http2-client/issues/79?

**Comment by serras** [comment]
@listx That's exactly the blocker
